gcp/opentelemetry-demo-workload-identity.yaml.gotmpl

```diff
@@ -23,3 +23,57 @@ opentelemetry-collector:
     annotations:
       "iam.gke.io/gcp-service-account": "opentelemetry-demo@{{ . }}.iam.gserviceaccount.com"
 {{ end }}
+
+{{ if .Values.collector_image_repository }}
+  image:
+{{ with .Values | getOrNil "collector_image_repository" }}
+    repository: "{{ . }}"
+{{ end }}
+{{ if .Values.collector_image_tag }}
+{{ with .Values | getOrNil "collector_image_tag" }}
+    tag: "{{ . }}"
+{{ end }}
+{{ end}}
+{{ end}}
+
+{{ if .Values.otlp_endpoint }}
+  config:
+    exporters:
+      otlphttp:
+{{ with .Values | getOrNil "otlp_endpoint" }}
+        endpoint: "{{ . }}"
+{{ end }}
+        encoding: json
+        auth:
+          authenticator: googleclientauth
+
+    processors:
+      resource:
+        attributes:
+        - action: insert
+          from_attribute: k8s.pod.uid
+          key: service.instance.id
+        - action: insert
+{{ with .Values | getOrNil "auth_extension_project_id" }}
+          value: "{{ . }}"
+{{ end }}
+          key: gcp.project_id
+
+    extensions:
+      googleclientauth:
+        scopes:
+        - "https://www.googleapis.com/auth/trace.append"
+        - "https://www.googleapis.com/auth/cloud-platform"
+{{ with .Values | getOrNil "auth_extension_project_id" }}
+        project: "{{ . }}"
+{{ end }}
+{{ with .Values | getOrNil "auth_extension_quota_project_id" }}
+        quota_project: "{{ . }}"
+{{ end }}
```

Comment:
Why do these project IDs need to be explicitly set for any of our deployment environments?

Reply:
we don't need them for our prod deployment, but other users may if they need to override the x-goog-user-project or x-goog-project-id headers required by the endpoint

Reply:
`auth_extension_project_id` sounds like it's misnamed. It sounds like it should just be nmws `destination_project_id` - it's the project that the data will be written to, right?

But the resource detector should already take care of detecting the current project.

It sounds like the OTLP endpoint currently requires `gcp.project_id` in addition to `cloud.account.id`. The endpoint really should default to the latter, but in the meantime we should just use the `resource` processor to copy the latter to the former.

The default quota project should be sufficient for this demo, so I don't think we need to do anything to configure at all. (It should default to either the project that the data is being written to or the project the service account comes from.)